core/src/main/java/org/keycloak/representations/idm/RealmRepresentation.java

```diff
@@ -598,6 +599,14 @@ public void setEditUsernameAllowed(Boolean editUsernameAllowed) {
         this.editUsernameAllowed = editUsernameAllowed;
     }
 
+    public Boolean isUserDeleteOwnAccountAllowed() {
+        return userDeleteOwnAccountAllowed;
+    }
+
+    public void setUserDeleteOwnAccountAllowed(Boolean userDeleteOwnAccountAllowed) {
+        this.userDeleteOwnAccountAllowed = userDeleteOwnAccountAllowed;
+    }
```

Comment:
Remove this - user should be allowed to delete account if user has delete role. Admin can enable/disable capability by deciding who gets the role, and enable/disable the delete action.